### PR TITLE
🔗 linter fix

### PR DESCRIPTION
**Pull request by trajepl**
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/engine/engine.py [hunk 1]
```diff
@@ -376,7 +376,7 @@ def setup_passes(self, accelerator_spec: "AcceleratorSpec"):
         # clean the passes
         self.passes.clear()
         for name, config in self.pass_config.items():
-            pass_cls: Type["Pass"] = config["type"]
```
Comment (guotuofeng) [inline olive/engine/engine.py:379]: will fail

Comment (guotuofeng) [reply olive/engine/engine.py:379]: https://stackoverflow.com/questions/77661139/how-do-i-stop-the-ruff-linter-from-moving-imports-into-a-type-checking-if-block

**Review by guotuofeng — CHANGES_REQUESTED** [review]
the typecheck  import need use double quotes

**Comment by trajepl** [comment]
> the typecheck import need use double quotes

In python, the annotions in function argument will be evaluated as a forward reference. Type annotations can be quoted to avoid forward references.

But when it comes to function variables, python always evaluate type annotations in a deferred manner, making the quotes unnecessary. https://docs.astral.sh/ruff/rules/quoted-annotation/

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**